subsys/bluetooth/host/a2dp.c

```diff
@@ -33,20 +51,173 @@
+	uint8_t pcm_callback_type_r;
+#endif
+	uint8_t sbc_first_data:1;
+	uint8_t pcm_first_data:1;
+};
+#endif
+
+struct bt_a2dp_encoder_decoder_state {
+#if defined(CONFIG_BT_A2DP_SOURCE)
+	struct bt_a2dp_encoder_sbc encoder_sbc;
+#endif
+#if defined(CONFIG_BT_A2DP_SINK)
+	struct bt_a2dp_decoder_sbc decoder_sbc;
+#endif
+};
+
+struct bt_a2dp_endpoint_internal {
+	/* the a2dp connection */
+	struct bt_a2dp *a2dp;
+	/* the corresponding endpoint */
+	struct bt_a2dp_endpoint *endpoint;
+	/* remote SEP Code Type */
+	uint8_t remote_codec_type;
+	/* the config result */
+	struct bt_a2dp_codec_ie_internal config_internal;
+	/* remote SEP info */
+	struct bt_avdtp_seid_info remote_ep_info;
+	/* remote SEP capability */
+	struct bt_a2dp_codec_ie_internal remote_ep_ie;
+	struct bt_a2dp_encoder_decoder_state *encoder_decoder;
+};
 
 struct bt_a2dp {
 	struct bt_avdtp session;
+	struct bt_avdtp_discover_params discover_param;
+	struct bt_avdtp_get_capabilities_params get_capabilities_param;
+	struct bt_avdtp_set_configuration_params set_configuration_param;
+	struct bt_avdtp_open_params open_param;
+	struct bt_avdtp_start_params start_param;
+	void (*configure_cb)(int err);
+	bt_a2dp_discover_peer_endpoint_cb_t discover_endpoint_cb;
+	struct bt_avdtp_seid_info peer_seids[CONFIG_BT_A2DP_MAX_ENDPOINT_COUNT];
+	uint8_t auto_select_endpoint_index;
+	uint8_t get_capabilities_index;
+	enum bt_a2dp_internal_state a2dp_state;
+	uint8_t peer_seids_count;
+	uint8_t auto_configure_enabled : 1;
 };
 
+static void a2dp_thread_entry(void);
+K_THREAD_DEFINE(a2dp_thread_id, CONFIG_BT_A2DP_TASK_STACK_SIZE, a2dp_thread_entry, NULL, NULL,
+		NULL, CONFIG_BT_A2DP_TASK_PRIORITY, 0, 0);
+struct bt_a2dp_connect_cb connect_cb;
+static struct bt_a2dp_endpoint_internal endpoint_internals[CONFIG_BT_A2DP_MAX_ENDPOINT_COUNT];
+K_MSGQ_DEFINE(a2dp_msgq, sizeof(struct a2dp_task_msg), A2DP_MSG_MAX_COUNT, 4);
+K_MUTEX_DEFINE(a2dp_mutex);
+
+#define A2DP_LOCK() k_mutex_lock(&a2dp_mutex, K_FOREVER)
+#define A2DP_UNLOCK() k_mutex_unlock(&a2dp_mutex)
+
 /* Connections */
 static struct bt_a2dp connection[CONFIG_BT_MAX_CONN];
 
-void a2d_reset(struct bt_a2dp *a2dp_conn)
+static int bt_a2dp_get_seid_caps(struct bt_a2dp *a2dp);
+static void a2dp_set_task_msg(void *parameter, uint8_t event);
+
+static struct bt_a2dp_endpoint_internal *bt_a2dp_get_endpoint_state(
+	struct bt_a2dp_endpoint *endpoint)
 {
-	(void)memset(a2dp_conn, 0, sizeof(struct bt_a2dp));
+	for (uint8_t index = 0; index < CONFIG_BT_A2DP_MAX_ENDPOINT_COUNT; ++index) {
+		if (endpoint_internals[index].endpoint == endpoint) {
+			return &endpoint_internals[index];
+		}
+	}
+	return NULL;
 }
 
-struct bt_a2dp *get_new_connection(struct bt_conn *conn)
+static uint8_t bt_a2dp_get_endpoints_count(void)
+{
+	uint8_t index;
+
+	for (index = 0; index < CONFIG_BT_A2DP_MAX_ENDPOINT_COUNT; index++) {
+		if (endpoint_internals[index].endpoint == NULL) {
+			break;
+		}
+	}
+	return index;
+}
+
+static void a2d_reset(struct bt_a2dp *a2dp)
```

Comment:
Only the prefix of this function is a2d_. Is it better to change it to a2dp_?

Reply:
probably yes

Reply:
Updated, Thanks.